### PR TITLE
[feat] Add claims API service

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

Add `Claim` service:
- Create a claim
- Retrieve a claim
- List all claims (paginated)
- Get the next page of claims
- Cancel a claim

Add associated `Claim`, `ClaimCollection` and `ClaimHistoryEntry` classes
Add associated `ClaimType` and `ClaimPaymentMethod` enums

# Testing

- Added unit tests, fixtures and cassettes as needed

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: EasyPost/Models/API/ClaimType.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /// <summary>
+    ///     An enum representing a damage claim type.
+    /// </summary>
+    public static readonly ClaimType Damage = new(1, "damage");
+
+    /// <summary>
+    ///     An enum representing a loss claim type.
+    /// </summary>
+    public static readonly ClaimType Loss = new(2, "loss");
```
Comment (ralexmatthews) [inline EasyPost/Models/API/ClaimType.cs:18]:
```
|   │ | ╷
────┼────
| | │ | -
```

**Review by ralexmatthews — APPROVED** [review]
LGTM 👍 lots of github linting comments on it tho lol

**Review by ralexmatthews — APPROVED** [review]
Sick 👌

File: EasyPost/Models/API/Claim.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        private string? _type { get; set; }
+#pragma warning restore SA1300
+
+        /// <summary>
+        ///     The <see cref="ClaimType"/> of the claim.
+        /// </summary>
+        [JsonIgnore]
+#pragma warning disable SA1101
+        public ClaimType? Type => _type == null ? null : ValueEnum.FromValue<ClaimType>(_type);
+#pragma warning restore SA1101
```
Comment (jchen293) [inline EasyPost/Models/API/Claim.cs:140]: I don't think we need to have a model class for this, we can just define it as a string.

File: EasyPost/Models/API/Claim.cs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        private string? _paymentMethod { get; set; }
+#pragma warning restore SA1300
+
+        /// <summary>
+        ///     The <see cref="ClaimPaymentMethod"/> of the claim.
+        /// </summary>
+#pragma warning disable SA1101
+        [JsonIgnore]
+        public ClaimPaymentMethod? PaymentMethod => _paymentMethod == null ? null : ValueEnum.FromValue<ClaimPaymentMethod>(_paymentMethod);
+#pragma warning disable SA1101
```
Comment (jchen293) [inline EasyPost/Models/API/Claim.cs:78]: I don't think we need to have a model class for this, we can just define it as a string.

File: EasyPost/Models/API/Claim.cs (hunk 2 above)
Comment (nwithan8) [reply EasyPost/Models/API/Claim.cs:140]: Since we are using the enum in the request, we want to remain consistent and use it in the response as well.

**Pull request merged**